Automattic/wp-calypso
# Improve the premium plugins purchase flow

Fixes https://github.com/Automattic/wp-calypso/issues/65499

#### Proposed Changes

This PR improves the overall flow for purchasing premium plugins by applying these changes:
- Avoid making requests to fetch the Atomic transfer status if there is already an ongoing request
- Hide the "Back to plugins" link if the plugin has not been installed yet, effectively allowing us to partially revert changes from https://github.com/Automattic/wp-calypso/pull/65293
- Change the link of the "Manage plugins" button for Limited Atomic sites since they cannot access `wp-admin/plugins.php`
- Simplify the loading screen in order to account for cases where the site was already Atomic before purchasing a plugin

#### Testing Instructions

- Use the Calypso live link below
- Switch to Simple site with a Free plan
- Go to Plugins
- Purchase a WooCommerce extension
- Make sure the loading screen goes through 4 steps and it doesn't display the "Back to plugins" button
- In the "Thank you" page, make sure the "Manage plugins" button redirects you to `/plugins/:plugin/:site`
- Make sure that the "WooCommerce" menu items have been updated
- Purchase any other WooCommerce extension
- Make sure the loading screen has now 1 single step since the site is already Atomic
- Switch to Simple site with a Pro plan
- Install a free plugin
- Make sure you only see one loading screen with 3 steps before being redirected to the "Thank you" page
- In the "Thank you" page, make sure the "Manage plugins" button redirects you to `wp-admin/plugins.php`
- Switch to a self-hosted Jetpack site
- Install a free plugin
- In the "Thank you" page, make sure the "Manage plugins" button is enabled and redirects you to `wp-admin/plugins.php`

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
